pkg/command/context.go

```diff
@@ -1438,6 +1448,9 @@ func displayContextListOutputWithDynamicColumns(cfg *configtypes.ClientConfig, w
 		dynamicColumns = append(dynamicColumns, "ClusterGroup")
 	}
 	if showAllColumns {
+		if tanzuContextExists {
+			requiredColumns = addProjectIDColumn(requiredColumns)
```

Comment:
I see what you are doing to insert the ProjectID column in the right place 😄 .
However, the `renderDynamicTable()` function has some magic that puts a column in the correct place based on the order of the `ContextListOutputRow` type.

So, you can simply change line 1448 to become `dynamicColumns = append(dynamicColumns, "ClusterGroup", "ProjectID")` and avoid having to do the gymnastics of inserting the column in the right place.

Reply:
Thanks @marckhouzam . I was not aware of this magic in `renderDynamicTable()`. Updated to use the `dynamicColumns` for projectID when `--wide` options is enabled.